provider/cmd/pulumi-gen-aws-native/main.go

```diff
@@ -73,8 +75,11 @@ func main() {
 
 	supportedTypes := readSupportedResourceTypes(genDir)
 	jsonSchemas := readJsonSchemas(schemaFolder)
-
-	fullSpec, _, err := schema.GatherPackage(supportedTypes, jsonSchemas, true)
+	semanticsDocument, err := schema.GatherSemantics(semanticsDir)
+	if err != nil {
+		panic(err)
```

Comment:
Nit: Worth wrapping similar to below?
```suggestion
		panic(fmt.Sprintf("error gathering semantics: %v", err))
```